ChiModules/LinearBoltzmannSolver/lbs_compute_balance.cc

```diff
@@ -0,0 +1,167 @@
+
+  //======================================== Get material source
+  // This is done using the SetSource routine
+  // because it allows a lot of flexibility.
+  auto mat_src = phi_old_local;
+  mat_src.assign(mat_src.size(),0.0);
+  for (auto& groupset : group_sets)
+  {
+    SetSource(groupset, APPLY_MATERIAL_SOURCE | APPLY_FISSION_SOURCE);
+    ScopedCopySTLvectors(groupset, q_moments_local, mat_src);
+  }
+
+  //======================================== Initialize diffusion params
+  //                                         for xs
+  // This populates sigma_a
+  for (auto& xs : material_xs)
+    if (not xs->diffusion_initialized)
+      xs->ComputeDiffusionParameters();
+
+  //======================================== Compute absorption, material-source
+  //                                         and in-flow
+  size_t num_groups=groups.size();
+  double local_out_flow   = 0.0;
+  double local_in_flow    = 0.0;
+  double local_absorption = 0.0;
+  double local_production = 0.0;
+  for (const auto& cell : grid->local_cells)
+  {
+    const auto&  transport_view   = cell_transport_views[cell.local_id];
+    const auto&  fe_intgrl_values = grid_fe_view.GetUnitIntegrals(cell);
+    const size_t num_nodes        = transport_view.NumNodes();
+    const auto&  IntV_shapeI      = fe_intgrl_values.GetIntV_shapeI();
+    const auto&  IntS_shapeI      = fe_intgrl_values.GetIntS_shapeI();
+
+    //====================================== Inflow
+    // This is essentially an integration over
+    // all faces, all angles, and all groups.
+    // Only the cosines that are negative are
+    // added to the integral.
+    for (int f=0; f<cell.faces.size(); ++f)
+    {
+      const auto& face  = cell.faces[f];
+
+      for (const auto& groupset : group_sets)
+      {
+        for (int n = 0; n < groupset.quadrature->omegas.size(); ++n)
+        {
+          const auto &omega = groupset.quadrature->omegas[n];
+          const double wt = groupset.quadrature->weights[n];
+          const double mu = omega.Dot(face.normal);
+
+          if (mu < 0.0 and (not face.has_neighbor)) //mu<0 and bndry
+          {
+            const auto &bndry = sweep_boundaries[face.neighbor_id];
+            for (int fi = 0; fi < face.vertex_ids.size(); ++fi)
+            {
+              const int i = fe_intgrl_values.FaceDofMapping(f, fi);
+              const auto &IntFi_shapeI = IntS_shapeI[f][i];
+
+              for (const auto &group : groupset.groups)
+              {
+                const int g = group.id;
+                const double psi = bndry->boundary_flux[g];
+                local_in_flow -= mu * wt * psi * IntFi_shapeI;
+              }//for g
+            }//for fi
+          }//if bndry
+        }//for n
+      }//for groupset
+    }//for f
+
+
+    //====================================== Outflow
+    //The group-wise outflow was determined
+    //during a solve so here we just
+    //consolidate it.
+    for (int g=0; g<num_groups; ++g)
```

Comment:
Same as above about size_t.

Reply:
Are we resolved on this one that we can leave those as `for (int`?